### PR TITLE
Feature: Add global filtering capabilities

### LINKED ISSUE #750: Extension: Global Uniform Filtering from Insights page
## Feature Request

**Description:**
Provide Uniform Filtering capabilities directly from the Insights page

### PR DESCRIPTION
**Pull request by mayan-000**
## Description
This PR adds global filtering capabilities to the landing/insights page of the extension, site, and sitemap report.
- The report downloaded will have a header showing applied filters and filtered matrix count.
- Whenever a filter is applied on the landing page it should update the count of matrices and filters should persist across tables.

## Testing Instructions
- Open the PSAT panel and navigate to the landing page.
- Select the filters and they should be applied to the matrix, and chips should be available.
- Now navigate to the table, the filters will be present with filtered data.
- Now download the report, it should also contain chips and filtered data.
- Now try in cli with available flags.

## Screenshot/Screencast

Screenshot 2024-07-09 at 17 03 21

---

## Checklist

- [x] I have thoroughly tested this code to the best of my abilities.
- [x] I have reviewed the code myself before requesting a review.
- [x] This code is covered by unit tests to verify that it works as intended.
- [x] The QA of this PR is done by a member of the QA team (to be checked by QA).

Fixes #750

### CODE REVIEW

File: packages/cli-dashboard/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   },
   "homepage": "https://github.com/GoogleChromeLabs/ps-analysis-tool#readme",
   "scripts": {
-    "dev": "webpack --config ../../dashboard.webpack.config.cjs --watch",
+    "dev": "NODE_ENV=development webpack serve --config ../../dashboard.webpack.config.cjs",
```
Comment (mohdsayed) [inline packages/cli-dashboard/package.json:27]:
```suggestion
    "dev": "webpack serve --config ../../dashboard.webpack.config.cjs  --open",
```

The default is `development`, so will `NODE_ENV=development` be required? Also, let's open it after serving.

https://github.com/GoogleChromeLabs/ps-analysis-tool/blob/2b89c2ef1685d1f5700d021b888a936cf8a0c6c4/webpack.shared.cjs#L19

File: packages/design-system/src/components/cookiesLanding/useGlobalFiltering.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies.
+ */
+import { useMemo } from 'react';
+import type { CookieTableData } from '@google-psat/common';
+import { I18n } from '@google-psat/i18n';
+import {
```
Comment (mohdsayed) [inline packages/design-system/src/components/cookiesLanding/useGlobalFiltering.tsx:23]:
```suggestion
/**
 * Internal dependencies
 */
import {
```

File: packages/extension/src/view/devtools/components/cookies/cookieLanding/index.tsx [hunk 3]
```diff
@@ -13,114 +13,102 @@
...
  */
-import CookiesSection from './cookiesSection';
-import FramesSection from './framesSection';
-import BlockedCookiesSection from './blockedCookiesSection';
-import { useCookie, useSettings } from '../../../stateProviders';
-import downloadReport from '../../../../../utils/downloadReport';
-import ExemptedCookiesSection from './exemptedCookiesSection';
+import Landing from './landing';
+import { useCookie } from '../../../stateProviders';
+import { Resizable } from 're-resizable';
```
Comment (mohdsayed) [inline packages/extension/src/view/devtools/components/cookies/cookieLanding/index.tsx:40]: Move to "External dependencies"

File: .eslintrc.json [hunk 4]
```diff
@@ -139,7 +139,8 @@
         "forbid": ["for"]
       }
     ],
-    "@typescript-eslint/ban-ts-comment": "off"
+    "@typescript-eslint/ban-ts-comment": "off",
+    "react/react-in-jsx-scope": "off"
```
Comment (mohdsayed) [inline .eslintrc.json:143]: This should have been done a separate PR.

Comment (mayan-000) [reply .eslintrc.json:143]: Library detection build was failing, that's why it was required here.

**Review by mohdsayed — APPROVED** [review]
Looks good!
